src/hooks/useServiceContainer.ts

```diff
@@ -85,8 +116,8 @@ export const useServiceContainer = (service: Function, loadingStateDelayMs: numb
     lastAbortController.current = new AbortController();
     requestConfig.signal = lastAbortController.current.signal;
 
-    return (serviceData ? service(serviceData, requestConfig) : service(requestConfig))
-      .then((response: any) => {
+    return (serviceData ? service(serviceData, requestConfig) : (service as ServiceFunctionWithoutParams<T>)(requestConfig))
```

Comment:
`(service as ServiceFunctionWithParams<T, U>)(serviceData, requestConfig)` ?

Reply:
Type cast is not required here.

Reply:
Not to compile successfully, but from logical and consistency point of view?

Reply:
That would not help anything.

Reply:
done